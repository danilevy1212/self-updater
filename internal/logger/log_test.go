package logger

import (
	"io"
	"os"
	"testing"
	"github.com/gkampitakis/go-snaps/snaps"
)

func Test_logger_New(t *testing.T) {
	t.Run("should log out pretty logs pretty = true", func(t *testing.T) {
		// Capture Stdout
		oldOut := os.Stdout
		r, w, _ := os.Pipe()
		os.Stdout = w

		l := New(true)

		if l == nil {
			t.Fatalf("Expected logger to be created, got nil")
		}

		l.Info().Msg("I'm supposed to be pretty")

		w.Close()

		msg, err := io.ReadAll(r)
		os.Stdout = oldOut

		if err != nil {
			t.Fatalf("Failed to read from pipe: %v", err)
		}

		snaps.MatchSnapshot(t, string(msg))
	})

	t.Run("should print out JSON logs pretty = false", func(t *testing.T) {
		// Capture Stdout
		oldOut := os.Stdout
		r, w, _ := os.Pipe()
		os.Stdout = w

		l := New(false)

		if l == nil {
			t.Fatalf("Expected logger to be created, got nil")
		}

		l.Info().Msg("I'm supposed to be JSON")
		w.Close()

		msg, err := io.ReadAll(r)
		os.Stdout = oldOut

		if err != nil {
			t.Fatalf("Failed to read from pipe: %v", err)
		}

		snaps.MatchSnapshot(t, string(msg))
	})

}
