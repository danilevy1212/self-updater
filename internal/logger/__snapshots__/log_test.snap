
[Test_logger_New/should_log_out_pretty_logs_pretty_=_true - 1]
[90m2025-03-27T05:00:00-07:00[0m [32mINF[0m [1mI'm supposed to be pretty[0m

---

[Test_logger_New/should_print_out_JSON_logs_pretty_=_false - 1]
{"level":"info","time":"2025-03-27T12:00:00Z","message":"I'm supposed to be JSON"}

---
